### ISSUE TITLE
`nix flake update` doesn't seem to work with multiple inputs

### ISSUE THREAD

**Issue by n8henrie** [NixOS/nix#9764]
**Describe the bug**

The [`nix flake update` docs](https://github.com/NixOS/nix/blob/cbd5553d57ebf5d0532047165a2d81825424bd76/src/nix/flake-update.md?plain=1#L50C1-L52C76) specify
> Unlike other `nix flake` commands, `nix flake update` takes a list of names of inputs
to update as its positional arguments and operates on the flake in the current directory.
You can pass a different flake-url with `--flake` to override that default.

but this doesn't seem to work.

**Steps To Reproduce**

```console
$ nix flake update nixpkgs-stable
warning: updating lock file '/Users/n8henrie/git/nixos/flake.lock':
• Updated input 'modules':
    'path:/nix/store/k89kr521x8cd49lkn68by7sxfgcy2kwj-source/modules?lastModified=1&narHash=sha256-UAaa9Fyj2JAbHGj7MoBMBv7Ym4aESGcTFeVEpm44QEA%3D' (1970-01-01)
  → 'path:/nix/store/k7ysr5jlwfv2gkqymlb3z9mdvi972ici-source/modules?lastModified=1&narHash=sha256-UAaa9Fyj2JAbHGj7MoBMBv7Ym4aESGcTFeVEpm44QEA%3D' (1970-01-01)
• Updated input 'overlays':
    'path:/nix/store/k89kr521x8cd49lkn68by7sxfgcy2kwj-source/overlays?lastModified=1&narHash=sha256-/7vpfCda1hJZ07BMdZhosGKYeHULvvhb2z7ZtEsA4SI%3D' (1970-01-01)
  → 'path:/nix/store/k7ysr5jlwfv2gkqymlb3z9mdvi972ici-source/overlays?lastModified=1&narHash=sha256-/7vpfCda1hJZ07BMdZhosGKYeHULvvhb2z7ZtEsA4SI%3D' (1970-01-01)
$ nix flake update home-manager
warning: updating lock file '/Users/n8henrie/git/nixos/flake.lock':
• Updated input 'modules':
    'path:/nix/store/k7ysr5jlwfv2gkqymlb3z9mdvi972ici-source/modules?lastModified=1&narHash=sha256-UAaa9Fyj2JAbHGj7MoBMBv7Ym4aESGcTFeVEpm44QEA%3D' (1970-01-01)
  → 'path:/nix/store/zwmfmxpysz7d84qr35rpnfl3h49v2g07-source/modules?lastModified=1&narHash=sha256-UAaa9Fyj2JAbHGj7MoBMBv7Ym4aESGcTFeVEpm44QEA%3D' (1970-01-01)
• Updated input 'overlays':
    'path:/nix/store/k7ysr5jlwfv2gkqymlb3z9mdvi972ici-source/overlays?lastModified=1&narHash=sha256-/7vpfCda1hJZ07BMdZhosGKYeHULvvhb2z7ZtEsA4SI%3D' (1970-01-01)
  → 'path:/nix/store/zwmfmxpysz7d84qr35rpnfl3h49v2g07-source/overlays?lastModified=1&narHash=sha256-/7vpfCda1hJZ07BMdZhosGKYeHULvvhb2z7ZtEsA4SI%3D' (1970-01-01)
$
$ nix flake update nixpkgs-stable home-manager
error: unexpected argument 'home-manager'
Try 'nix --help' for more information.
$ nix flake update nixpkgs-stable,home-manager
warning: Invalid flake input 'nixpkgs-stable,home-manager'. To update a specific flake, use 'nix flake update --flake nixpkgs-stable,home-manager' instead.
error: invalid flake input path element 'nixpkgs-stable,home-manager'
$ nix flake update 'nixpkgs-stable home-manager'
warning: Invalid flake input 'nixpkgs-stable home-manager'. To update a specific flake, use 'nix flake update --flake nixpkgs-stable home-manager' instead.
error: invalid flake input path element 'nixpkgs-stable home-manager'
$ nix flake update '["nixpkgs-stable" "home-manager"]'
warning: Invalid flake input '["nixpkgs-stable" "home-manager"]'. To update a specific flake, use 'nix flake update --flake ["nixpkgs-stable" "home-manager"]' instead.
error: invalid flake input path element '["nixpkgs-stable" "home-manager"]'
```
Am I holding it wrong?

**Expected behavior**

```console
$ nix flake update nixpkgs-stable home-manager
... works ...
```

**`nix-env --version` output**

```console
$ nix-env --version
nix-env (Nix) 2.19.2
```

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by roberth** [comment]
> Am I holding it wrong?

No, I would expect the same.

**Comment by thufschmitt** [comment]
(continuing the discussion from #10032 where @olmokramer offered to have a stab a implementing it)

Thanks @olmokramer, you're definitely welcome to try!

I suspect that it's a fairly simple fix (but might require some digging into the internals). Mostly it should be about understanding why

https://github.com/NixOS/nix/blob/60936f28e51bad1cf0a038ebfa24ce9bb97845a9/src/nix/flake.cc#L88-L106

(the code that defines the arguments to `nix flake update`) doesn't behave like

https://github.com/NixOS/nix/blob/60936f28e51bad1cf0a038ebfa24ce9bb97845a9/src/nix/flake.cc#L38-L45

(the code that defines the arguments for the other flake commands)

**Comment by olmokramer** [comment]
@thufschmitt Thanks for the pointers! I think I've figured it out: the `handler` of `CmdFlakeUpdate` only accepts a single `std::string` instead of a `std::vector<std::string>` and so would only read a single argument. After processing the first argument the `expectedArgs` list would be empty while there were still more command line arguments to be processed, and so it would throw the error. I've opened a PR at https://github.com/NixOS/nix/pull/10073 which seems to fix it: I ran it on the `nix` repository itself with `./outputs/out/bin/nix flake update nixpkgs libgit2` and indeed it updated both inputs!
